merlin/models/tf/data_augmentation/negative_sampling.py

```diff
@@ -52,7 +52,7 @@ def __init__(
         n_per_positive: int,
         seed: Optional[int] = None,
         run_when_testing: bool = True,
-        return_tuple: bool = False,
+        return_tuple: bool = True,
```

Comment:
In what cases does it matter whether it’s a standard tuple versus our `Prediction`-object which is a NamedTuple.

Reply:
This matters when this sampling layer is used as an argument to `BatchedDataset.map(...)`. When returning the Prediction NamedTuple (`return_tuple=False`) and passed to a model.fit an exception is raised from keras. Reported in #626 

```
ValueError: Data is expected to be in format `x`, `(x,)`, `(x, y)`, or `(x, y, sample_weight)`, found: Prediction
```

This `return_tuple` flag was added to work around this issue. However, it doesn't feel like the most natural thing to use.

The proposal in this PR is to change the default so that the supported use-case (in the dataloader) works without requiring this parameter to be specified. If we want to support running this as part of the model, we should aim to find a way for it to work without requiring this parameter.

Reply:
I would prefer to keep the default of this flag `False` and to put in a small PR to implement our own version of `unpack_x_y_sample_weight` (and make use of that in [train_step](https://github.com/NVIDIA-Merlin/models/blob/main/merlin/models/tf/models/base.py#L580) & [test_step](https://github.com/NVIDIA-Merlin/models/blob/main/merlin/models/tf/models/base.py#L598))

Reply:
I've tried out the idea of updating `train_step` and `test_step`  in #702 . However, it seems that inside the keras model there are methods we're calling on the keras model base class that use  `unpack_x_y_sample_weight`. So updating `train_step` and `test_step` is not enough. I guess monkey-patching the function might work around this, though doesn't feel like the preferred solution.

Reply:
An alternative that borrows from this idea, is to use the fact that our `train_step` and `test_step` pass in the parameters `training` and `testing`. We can use those to figure out if the layer is being called from a Merlin Model. And figure out the return type based on those.

Implemented this idea in PR #703